do-core/src/instruction.rs

```diff
@@ -21,6 +21,19 @@ impl OpCode {
     }
 }
 
+impl std::str::FromStr for OpCode {
+    type Err = Error;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "LD"  => Ok(OpCode::LD),
+            "ST"  => Ok(OpCode::ST),
+            "ADD" => Ok(OpCode::ADD),
+            "XOR" => Ok(OpCode::XOR),
+            _ =>     Err(Error::ParseOpError),
```

Comment:
This needs some formatting love. `cargo fmt` will help you.

Reply:
i ran it earlier and it unaligns everything even more :
```rust
            "LD" => Ok(OpCode::LD),
            "ST" => Ok(OpCode::ST),
            "ADD" => Ok(OpCode::ADD),
            "XOR" => Ok(OpCode::XOR),
            _ => Err(Error::ParseOpError),
```